### PR TITLE
keras text processing layer

### PR DESCRIPTION
**Pull request by BenoitDherin**
This PR adds an optional section to the Keras text classification lab solution. This optional section builds and trains a RNN using the new Keras text preprocessing layer directly as input layer.

### CODE REVIEW

File: notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb [hunk 1]
```diff
@@ -910,6 +910,218 @@
     "cnn_model.summary()"
```
Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>When running through this lab, the dnn_history consisted of only 1 point. I needed to change PATIENCE=2 to get reasonable plots for train/validation loss. This is new behavior from having run the labs recently, which is odd, but might consider changing PATIENCE=0 to PATIENCE=5.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Similarly here. I changed PATIENCE=2 and modified the learning rate to be 0.0001 at compile step setting <code>optimizer=tf.keras.optimizers.Adam(learning_rate=0.0001)</code>. Takes a bit longer to train but much nicer train/validation loss curves.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='66'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>As above. I changed PATIENCE=2 and modified the learning rate to be 0.0001 at compile step setting&nbsp;<code>optimizer=tf.keras.optimizers.Adam(learning_rate=0.0001)</code>. Takes a bit longer to train but much nicer train/validation loss curves.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='71'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>typo: maxium -&gt; maximum</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='75'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>nit: predicts -&gt; predict</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='88'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>UNITS hasn't been defined yet at this point.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='89'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>same here. I'd change the PATIENCE to 2 and learning rate to 0.0001, but just to get better plots below. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='91'/>

Comment (munnm) [inline notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>make Copyright 2022?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='92'/>

**Comment by munnm** [comment]
This is great Benoit! Thanks for the quick turn around. Just a couple nits/suggestions. Looks to work well and as expected :) Thanks!

File: notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb (hunk 1 above)
Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>I think it has on line 11 in the same cell, hasn't it?</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>

Comment (BenoitDherin) [reply notebooks/text_classification/solutions/2_keras_for_text_classification.ipynb:910]: <p>Done!</p>
